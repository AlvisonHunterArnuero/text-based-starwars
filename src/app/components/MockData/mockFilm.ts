export const mockFilm =
{
  title: 'A New Hope',
  episode_id: 4,
  release_date: '1977-05-25',
  opening_crawl: 'It is a period of civil war. Rebel spaceships...',
  director: 'George Lucas',
  producer: 'Gary Kurtz, Rick McCallum',
  characters: Array(10).fill('character'),
  planets: Array(5).fill('planet'),
  starships: Array(3).fill('starship'),
  species: Array(4).fill('species'),
  vehicles: Array(2).fill('vehicle'),
  created: '1977-05-01T00:00:00.000Z',
  edited: '1977-05-02T00:00:00.000Z',
  url: 'https://swapi.dev/api/films/1/',
};