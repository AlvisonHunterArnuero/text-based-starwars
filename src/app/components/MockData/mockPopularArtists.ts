import { TMDBPopularArtist } from "@/types/tmdb";

export const mockArtists: TMDBPopularArtist[] = [
    {
        id: 1,
        name: 'Scarlett Johansson',
        original_name: 'Scarlett Johansson',
        popularity: 85.5,
        gender: 1,
        known_for_department: 'Acting',
        profile_path: '/test-profile1.jpg',
        known_for: [
            {
                id: 101,
                title: 'Lucy',
                name: 'Series 1',
                poster_path: '/poster1.jpg',
                media_type: 'movie',
                backdrop_path: null,
                overview: '',
                adult: false,
                original_language: '',
                genre_ids: [],
                popularity: 0,
                vote_average: 0,
                vote_count: 0
            },
        ],
        adult: false,
        backdrop_path: null
    },
    {
        id: 2,
        name: 'Sandra Bullock',
        original_name: 'Sandra Bullock',
        popularity: 72.3,
        gender: 2,
        known_for_department: 'Directing',
        profile_path: '/test-profile2.jpg',
        known_for: [
            {
                id: 102,
                title: 'Gravity',
                name: 'Series 2',
                poster_path: '/poster2.jpg',
                media_type: 'tv',
                backdrop_path: null,
                overview: '',
                adult: false,
                original_language: '',
                genre_ids: [],
                popularity: 0,
                vote_average: 0,
                vote_count: 0
            },
        ],
        adult: false,
        backdrop_path: null
    },
];