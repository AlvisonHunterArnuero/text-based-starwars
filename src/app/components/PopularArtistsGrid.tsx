/* eslint-disable @next/next/no-img-element */
import FemaleIcon from '@mui/icons-material/Female';
import InfoIcon from '@mui/icons-material/Info';
import MaleIcon from '@mui/icons-material/Male';
import {
  Grid,
  Card,
  CardMedia,
  CardContent,
  Typography,
  CardActions,
  Chip,
  Box,
  Stack,
  Container,
} from '@mui/material';
import IconButton from '@mui/material/IconButton';
import ImageList from '@mui/material/ImageList';
import ImageListItem from '@mui/material/ImageListItem';
import ImageListItemBar from '@mui/material/ImageListItemBar';
import ListSubheader from '@mui/material/ListSubheader';
import React from 'react';

import { TMDBPopularArtist } from '@/types/tmdb';

import { fillMovieAverageByRateValue } from '../../lib/utils';

interface MovieGridProps {
  movies: TMDBPopularArtist[];
}

const posterThumbsPath = process.env.NEXT_PUBLIC_POSTER_THUMBS_PATH;
const stillThumbsPath = process.env.NEXT_PUBLIC_STILL_THUMBS_PATH;

const PopularArtistsGrid: React.FC<MovieGridProps> = ({ movies }) => {
  if (!movies || movies.length === 0) {
    return (
      <Box sx={{ p: 3, textAlign: 'center' }}>
        <Typography variant="h4" color="error">
          No movies found with that name.
        </Typography>
      </Box>
    );
  }

  return (
    <Box
      sx={{
        flexGrow: 1,
        p: 2,
        backgroundColor: '#212121',
        borderRadius: 4,
      }}
    >
      <Typography color="white" variant="h4" gutterBottom>
        Popular Artists
      </Typography>
      <Typography
        variant="subtitle1"
        color="warning.main"
        gutterBottom
      >
        Discover the most popular artists in the film industry.
      </Typography>
      <Container fixed>
        {movies.map((artist) => (
          <Grid
            size={12}
            overflow={'hidden'}
            key={artist.id}
            border={1}
            borderRadius={2}
            sx={{
              maxWidth: 2000,
              gap: 2,
              mb: 3,
            }}
          >
            <Card
              role="article"
              data-testid={`artist-card-${artist.id}`}
              sx={{
                height: '100%',
                display: 'flex',
                flexDirection: 'row',
                transition: 'transform 0.2s',
                '&:hover': {
                  transform: 'scale(1.02)',
                  boxShadow: (theme) => theme.shadows[5],
                },
              }}
            >
              <CardMedia
                component="img"
                height="200"
                image={`${stillThumbsPath}${artist.profile_path}`}
                alt={artist.name}
                sx={{
                  objectFit: 'cover',
                  maxWidth: 700,
                  minWidth: 300,
                  backgroundColor: (theme) => theme.palette.grey[500],
                }}
              />

              <CardContent sx={{ pt: 2, flexGrow: 1 }}>
                <Stack
                  display={'flex'}
                  flexDirection={'row'}
                  gap={1}
                  justifyContent={'space-between'}
                  alignItems={'center'}
                  pb={1}
                >
                  <Typography
                    variant="h5"
                    component="div"
                    color="primary.main"
                  >
                    {artist.name}
                  </Typography>
                  <Chip
                    label={artist.known_for_department}
                    variant="outlined"
                  />
                </Stack>
                <Typography
                  variant="subtitle1"
                  component="div"
                  color="secondary.main"
                >
                  Real Name: {artist.original_name}
                </Typography>

                <CardActions
                  sx={{
                    display: 'flex',
                    justifyContent: 'space-between',
                    p: 2,
                    borderTop: `1px solid #333`,
                    alignItems: 'center',
                    border: '1px solid rgba(255, 255, 255, 0.9)',
                  }}
                >
                  <Box sx={{ display: 'flex', gap: 1, marginTop: 1 }}>
                    <Chip
                      size="small"
                      label="GENDER"
                      icon={
                        artist.gender == 1 ? (
                          <MaleIcon />
                        ) : (
                          <FemaleIcon />
                        )
                      }
                      variant="filled"
                    />
                    <Chip
                      label={`${artist.popularity.toFixed(1)} â˜…`}
                      size="small"
                      color={`${fillMovieAverageByRateValue(
                        artist.popularity.toFixed(1)
                      )}`}
                    />
                    <Chip
                      label={`Popularity: ${artist.popularity.toFixed(
                        0
                      )}`}
                      size="small"
                      color="secondary"
                      variant="outlined"
                    />
                  </Box>
                </CardActions>

                <Grid container spacing={4}>
                  <ImageList gap={8} cols={2}>
                    <ImageListItem key="Subheader" cols={2}>
                      <ListSubheader component="div">
                        <Typography
                          variant="h6"
                          color="text.secondary"
                          sx={{
                            display: '-webkit-box',
                            WebkitLineClamp: 3,
                            fontFamily: 'Inter, sans-serif',
                            WebkitBoxOrient: 'vertical',
                          }}
                        >
                          Known For:
                        </Typography>
                      </ListSubheader>
                    </ImageListItem>
                    {artist.known_for.map((item) => (
                      <ImageListItem cols={3} key={item.id}>
                        <img
                          src={`${posterThumbsPath}${item.poster_path}`}
                          alt={item.title || 'Unknown Title'}
                          loading="lazy"
                          width={248}
                          height={350}
                        />
                        <ImageListItemBar
                          title={item.title}
                          subtitle={item.name}
                          actionIcon={
                            <IconButton
                              sx={{
                                color: 'rgba(255, 255, 255, 0.54)',
                              }}
                              aria-label={`info about ${item.title}`}
                            >
                              <InfoIcon />
                            </IconButton>
                          }
                        />
                      </ImageListItem>
                    ))}
                  </ImageList>
                </Grid>
              </CardContent>
            </Card>
          </Grid>
        ))}
      </Container>
    </Box>
  );
};

export default PopularArtistsGrid;
