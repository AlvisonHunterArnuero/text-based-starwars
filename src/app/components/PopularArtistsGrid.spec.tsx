import { render, screen, within } from '@testing-library/react';
import React from 'react';
import { describe, it, expect } from 'vitest';

import { mockArtists } from './MockData/mockPopularArtists';
import PopularArtistsGrid from './PopularArtistsGrid';
import '@testing-library/jest-dom/vitest';

describe('PopularArtistsGrid Component', () => {
  it('renders without crashing', () => {
    render(<PopularArtistsGrid movies={mockArtists} />);
    expect(screen.getByText('Popular Artists')).toBeInTheDocument();
  });

  it('displays "No movies found" when empty array is passed', () => {
    render(<PopularArtistsGrid movies={[]} />);
    expect(
      screen.getByText('No movies found with that name.')
    ).toBeInTheDocument();
  });

  it('renders correct number of artist cards', () => {
    render(<PopularArtistsGrid movies={mockArtists} />);
    const cards = screen.getAllByRole('article');
    expect(cards).toHaveLength(mockArtists.length);
  });

  it('displays correct artist information', () => {
    render(<PopularArtistsGrid movies={mockArtists} />);

    // Test first artist
    expect(
      screen.getByText('Scarlett Johansson')
    ).toBeInTheDocument();
    expect(
      screen.getByText('Scarlett Johansson')
    ).toBeInTheDocument();
    expect(screen.getByText('Acting')).toBeInTheDocument();
    expect(screen.getByText('85.5 â˜…')).toBeInTheDocument();

    // Test second artist
    expect(screen.getByText('Sandra Bullock')).toBeInTheDocument();
    expect(screen.getByText('Directing')).toBeInTheDocument();
  });

  it('displays correct gender icon', () => {
    render(<PopularArtistsGrid movies={mockArtists} />);

    // Male icon for gender 1
    const maleChip = screen
      .getAllByText('GENDER')[0]
      .closest('.MuiChip-root');
    expect(maleChip).toHaveTextContent('GENDER');
    expect(maleChip?.querySelector('svg')).toBeInTheDocument(); // MaleIcon
  });

  it('renders known for items correctly', () => {
    render(<PopularArtistsGrid movies={mockArtists} />);

    const knownForSections = screen.getAllByText('Known For:');
    expect(knownForSections).toHaveLength(mockArtists.length);

    const firstArtistKnownFor = within(
      knownForSections[0].closest('.MuiImageList-root')!
    );
    expect(
      firstArtistKnownFor.getByAltText('Lucy')
    ).toBeInTheDocument();
  });

  it('applies correct styling to cards', () => {
    render(<PopularArtistsGrid movies={mockArtists} />);

    const firstCard = screen.getAllByRole('article')[0];
    expect(firstCard).toHaveStyle('display: flex');
    expect(firstCard).toHaveStyle('flex-direction: row');
  });

  it('matches snapshot with mock data', () => {
    const { container } = render(
      <PopularArtistsGrid movies={mockArtists} />
    );
    expect(container).toMatchSnapshot();
  });
});
