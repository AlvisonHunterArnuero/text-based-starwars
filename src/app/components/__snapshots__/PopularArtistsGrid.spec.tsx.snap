// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PopularArtistsGrid Component > matches snapshot with mock data 1`] = `
<div>
  <div
    class="MuiBox-root css-1cksp76"
  >
    <h4
      class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1fvtwfk-MuiTypography-root"
    >
      Popular Artists
    </h4>
    <h6
      class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom css-10jmx0n-MuiTypography-root"
    >
      Discover the most popular artists in the film industry.
    </h6>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg MuiContainer-fixed css-1dgpa8w-MuiContainer-root"
    >
      <div
        class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-12 css-dqwztt-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-14tng7z-MuiPaper-root-MuiCard-root"
          data-testid="artist-card-1"
          role="article"
          style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
        >
          <img
            alt="Scarlett Johansson"
            class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-1q6sk57-MuiCardMedia-root"
            height="200"
            src="undefined/test-profile1.jpg"
          />
          <div
            class="MuiCardContent-root css-1nfy826-MuiCardContent-root"
          >
            <div
              class="MuiStack-root css-abc9gl-MuiStack-root"
            >
              <div
                class="MuiTypography-root MuiTypography-h5 css-82crf8-MuiTypography-root"
              >
                Scarlett Johansson
              </div>
              <div
                class="MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-outlinedDefault css-1lrxra9-MuiChip-root"
              >
                <span
                  class="MuiChip-label MuiChip-labelMedium css-16cgrcw-MuiChip-label"
                >
                  Acting
                </span>
              </div>
            </div>
            <div
              class="MuiTypography-root MuiTypography-subtitle1 css-d8p31k-MuiTypography-root"
            >
              Real Name: 
              Scarlett Johansson
            </div>
            <div
              class="MuiCardActions-root MuiCardActions-spacing css-11sikwc-MuiCardActions-root"
            >
              <div
                class="MuiBox-root css-1uthhvt"
              >
                <div
                  class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault css-8aqjdy-MuiChip-root"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-icon MuiChip-iconSmall MuiChip-iconColorDefault css-1umw9bq-MuiSvgIcon-root"
                    data-testid="MaleIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M9.5 11c1.93 0 3.5 1.57 3.5 3.5S11.43 18 9.5 18 6 16.43 6 14.5 7.57 11 9.5 11m0-2C6.46 9 4 11.46 4 14.5S6.46 20 9.5 20s5.5-2.46 5.5-5.5c0-1.16-.36-2.23-.97-3.12L18 7.42V10h2V4h-6v2h2.58l-3.97 3.97C11.73 9.36 10.66 9 9.5 9"
                    />
                  </svg>
                  <span
                    class="MuiChip-label MuiChip-labelSmall css-eccknh-MuiChip-label"
                  >
                    GENDER
                  </span>
                </div>
                <div
                  class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorPrimary MuiChip-filledPrimary css-1abuvwb-MuiChip-root"
                >
                  <span
                    class="MuiChip-label MuiChip-labelSmall css-eccknh-MuiChip-label"
                  >
                    85.5 â˜…
                  </span>
                </div>
                <div
                  class="MuiChip-root MuiChip-outlined MuiChip-sizeSmall MuiChip-colorSecondary MuiChip-outlinedSecondary css-e0vu2d-MuiChip-root"
                >
                  <span
                    class="MuiChip-label MuiChip-labelSmall css-19m61dl-MuiChip-label"
                  >
                    Popularity: 86
                  </span>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row MuiGrid-spacing-xs-4 css-kqg16c-MuiGrid-root"
            >
              <ul
                class="MuiImageList-root MuiImageList-standard css-lmqlpj-MuiImageList-root"
                style="grid-template-columns: repeat(2, 1fr); gap: 8px;"
              >
                <li
                  class="MuiImageListItem-root MuiImageListItem-standard css-xjohuq-MuiImageListItem-root"
                  style="height: auto; grid-column-end: span 2; grid-row-end: span 1;"
                >
                  <div
                    class="MuiListSubheader-root MuiListSubheader-gutters MuiListSubheader-sticky css-l328gy-MuiListSubheader-root"
                  >
                    <h6
                      class="MuiTypography-root MuiTypography-h6 css-hqli7f-MuiTypography-root"
                    >
                      Known For:
                    </h6>
                  </div>
                </li>
                <li
                  class="MuiImageListItem-root MuiImageListItem-standard css-xjohuq-MuiImageListItem-root"
                  style="height: auto; grid-column-end: span 3; grid-row-end: span 1;"
                >
                  <img
                    alt="Lucy"
                    class="MuiImageListItem-img"
                    height="350"
                    loading="lazy"
                    src="undefined/poster1.jpg"
                    width="248"
                  />
                  <div
                    class="MuiImageListItemBar-root MuiImageListItemBar-positionBottom MuiImageListItemBar-actionPositionRight css-hbl0a4-MuiImageListItemBar-root"
                  >
                    <div
                      class="MuiImageListItemBar-titleWrap MuiImageListItemBar-titleWrapBottom MuiImageListItemBar-titleWrapActionPosRight css-xn78qt-MuiImageListItemBar-titleWrap"
                    >
                      <div
                        class="MuiImageListItemBar-title css-1sdhmys-MuiImageListItemBar-title"
                      >
                        Lucy
                      </div>
                      <div
                        class="MuiImageListItemBar-subtitle css-12mqd3t-MuiImageListItemBar-subtitle"
                      >
                        Series 1
                      </div>
                    </div>
                    <div
                      class="MuiImageListItemBar-actionIcon MuiImageListItemBar-actionIconActionPosRight css-6nlpuz-MuiImageListItemBar-actionIcon"
                    >
                      <button
                        aria-label="info about Lucy"
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-10u7adb-MuiButtonBase-root-MuiIconButton-root"
                        tabindex="0"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                          data-testid="InfoIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m1 15h-2v-6h2zm0-8h-2V7h2z"
                          />
                        </svg>
                      </button>
                    </div>
                  </div>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-12 css-dqwztt-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-14tng7z-MuiPaper-root-MuiCard-root"
          data-testid="artist-card-2"
          role="article"
          style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
        >
          <img
            alt="Sandra Bullock"
            class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-1q6sk57-MuiCardMedia-root"
            height="200"
            src="undefined/test-profile2.jpg"
          />
          <div
            class="MuiCardContent-root css-1nfy826-MuiCardContent-root"
          >
            <div
              class="MuiStack-root css-abc9gl-MuiStack-root"
            >
              <div
                class="MuiTypography-root MuiTypography-h5 css-82crf8-MuiTypography-root"
              >
                Sandra Bullock
              </div>
              <div
                class="MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-outlinedDefault css-1lrxra9-MuiChip-root"
              >
                <span
                  class="MuiChip-label MuiChip-labelMedium css-16cgrcw-MuiChip-label"
                >
                  Directing
                </span>
              </div>
            </div>
            <div
              class="MuiTypography-root MuiTypography-subtitle1 css-d8p31k-MuiTypography-root"
            >
              Real Name: 
              Sandra Bullock
            </div>
            <div
              class="MuiCardActions-root MuiCardActions-spacing css-11sikwc-MuiCardActions-root"
            >
              <div
                class="MuiBox-root css-1uthhvt"
              >
                <div
                  class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault css-8aqjdy-MuiChip-root"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-icon MuiChip-iconSmall MuiChip-iconColorDefault css-1umw9bq-MuiSvgIcon-root"
                    data-testid="FemaleIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M17.5 9.5C17.5 6.46 15.04 4 12 4S6.5 6.46 6.5 9.5c0 2.7 1.94 4.93 4.5 5.4V17H9v2h2v2h2v-2h2v-2h-2v-2.1c2.56-.47 4.5-2.7 4.5-5.4m-9 0C8.5 7.57 10.07 6 12 6s3.5 1.57 3.5 3.5S13.93 13 12 13s-3.5-1.57-3.5-3.5"
                    />
                  </svg>
                  <span
                    class="MuiChip-label MuiChip-labelSmall css-eccknh-MuiChip-label"
                  >
                    GENDER
                  </span>
                </div>
                <div
                  class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorPrimary MuiChip-filledPrimary css-1abuvwb-MuiChip-root"
                >
                  <span
                    class="MuiChip-label MuiChip-labelSmall css-eccknh-MuiChip-label"
                  >
                    72.3 â˜…
                  </span>
                </div>
                <div
                  class="MuiChip-root MuiChip-outlined MuiChip-sizeSmall MuiChip-colorSecondary MuiChip-outlinedSecondary css-e0vu2d-MuiChip-root"
                >
                  <span
                    class="MuiChip-label MuiChip-labelSmall css-19m61dl-MuiChip-label"
                  >
                    Popularity: 72
                  </span>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row MuiGrid-spacing-xs-4 css-kqg16c-MuiGrid-root"
            >
              <ul
                class="MuiImageList-root MuiImageList-standard css-lmqlpj-MuiImageList-root"
                style="grid-template-columns: repeat(2, 1fr); gap: 8px;"
              >
                <li
                  class="MuiImageListItem-root MuiImageListItem-standard css-xjohuq-MuiImageListItem-root"
                  style="height: auto; grid-column-end: span 2; grid-row-end: span 1;"
                >
                  <div
                    class="MuiListSubheader-root MuiListSubheader-gutters MuiListSubheader-sticky css-l328gy-MuiListSubheader-root"
                  >
                    <h6
                      class="MuiTypography-root MuiTypography-h6 css-hqli7f-MuiTypography-root"
                    >
                      Known For:
                    </h6>
                  </div>
                </li>
                <li
                  class="MuiImageListItem-root MuiImageListItem-standard css-xjohuq-MuiImageListItem-root"
                  style="height: auto; grid-column-end: span 3; grid-row-end: span 1;"
                >
                  <img
                    alt="Gravity"
                    class="MuiImageListItem-img"
                    height="350"
                    loading="lazy"
                    src="undefined/poster2.jpg"
                    width="248"
                  />
                  <div
                    class="MuiImageListItemBar-root MuiImageListItemBar-positionBottom MuiImageListItemBar-actionPositionRight css-hbl0a4-MuiImageListItemBar-root"
                  >
                    <div
                      class="MuiImageListItemBar-titleWrap MuiImageListItemBar-titleWrapBottom MuiImageListItemBar-titleWrapActionPosRight css-xn78qt-MuiImageListItemBar-titleWrap"
                    >
                      <div
                        class="MuiImageListItemBar-title css-1sdhmys-MuiImageListItemBar-title"
                      >
                        Gravity
                      </div>
                      <div
                        class="MuiImageListItemBar-subtitle css-12mqd3t-MuiImageListItemBar-subtitle"
                      >
                        Series 2
                      </div>
                    </div>
                    <div
                      class="MuiImageListItemBar-actionIcon MuiImageListItemBar-actionIconActionPosRight css-6nlpuz-MuiImageListItemBar-actionIcon"
                    >
                      <button
                        aria-label="info about Gravity"
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-10u7adb-MuiButtonBase-root-MuiIconButton-root"
                        tabindex="0"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                          data-testid="InfoIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m1 15h-2v-6h2zm0-8h-2V7h2z"
                          />
                        </svg>
                      </button>
                    </div>
                  </div>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
